src/coreclr/dlls/mscorrc/mscorrc.rc

```diff
@@ -724,28 +724,28 @@ END
-        IDS_PROF_PROFILER_DISABLED              "Profilers will not be loaded by the current version of the CLR because of the environment variable setting below.  Please consult the documentation for information on how to use this environment variable, and the risks associated with it.  Environment variable setting: %s=%s. Profiler CLSID: '%s'.\n"
+        IDS_PROF_CANCEL_ACTIVATION              "The profiler has requested that the CLR instance not load the profiler into this process.  Profiler GUID: '%s'.\n"
+        IDS_PROF_V2PROFILER_DISABLED            "Loading profiler failed.   The profiler that was configured to load was designed for an older version of the CLR.  You can use the DOTNET_ProfAPI_ProfilerCompatibilitySetting environment variable to allow older profilers to be loaded by the current version of the CLR.  Please consult the documentation for information on how to use this environment variable, and the risks associated with it.  Profiler GUID: '%s'.\n"
+        IDS_PROF_V2PROFILER_ENABLED             "A profiler designed for an older version of the CLR was loaded because of the environment variable setting below.  Older profilers will continue to work in many cases, but if you encounter problems, please consider upgrading the profiler or changing the setting of the environment variable.  Please consult the documentation for information on how to use this environment variable, and the risks associated with it.  Environment variable setting: %s=%s.  Profiler GUID: '%s'.\n"
+        IDS_PROF_PROFILER_DISABLED              "Profilers will not be loaded by the current version of the CLR because of the environment variable setting below.  Please consult the documentation for information on how to use this environment variable, and the risks associated with it.  Environment variable setting: %s=%s. Profiler GUID: '%s'.\n"
         IDS_E_PROF_NOTIFICATION_DISABLED        "Profiler was prevented from loading notification profiler due to app settings.\n"
         IDS_E_PROF_NOTIFICATION_LIMIT_EXCEEDED  "Notification profiler was prevented from loading because the limit of notification profilers was reached.\n"
-        IDS_E_PROF_TIMEOUT_WAITING_FOR_CONCURRENT_GC    "Profiler timed out on waiting for concurrent GC to finish after '%d' milliseconds. Please configure your profiler to increase its attaching time out value or consult the documentation for the COMPlus_ProfAPI_AttachProfilerMinTimeoutInMs environment variable and try again. Profiler CLSID: '%s'.\n"
+        IDS_E_PROF_TIMEOUT_WAITING_FOR_CONCURRENT_GC    "Profiler timed out on waiting for concurrent GC to finish after '%d' milliseconds. Please configure your profiler to increase its attaching time out value or consult the documentation for the DOTNET_ProfAPI_AttachProfilerMinTimeoutInMs environment variable and try again. Profiler GUID: '%s'.\n"
```

Comment:
The `*_ProfAPI_AttachProfilerMinTimeoutInMs` configuration knob doesn't seem to exist anymore. Which is really weird because this error message is used. Lots of confusion here. @davmason Can you share some guidance here?

Reply:
The timeout is hardcoded to be INFINITE. This error message and the code related to it is unreachable.

Reply:
Actually, the timeout can be specified via the event pipe diagnostic protocol: https://github.com/dotnet/runtime/blob/c4d5f6dff367ba1197358e5abd0d50d8b10a3fef/src/coreclr/vm/eventing/eventpipe/ds-rt-coreclr.h#L291

Reply:
I removed the unreachable code.

Reply:
I am sorry my earlier comment about the timeout being unreachable was not correct. The timeout is reachable and can be specified by the event pipe diagnostic protocol (`ds_attach_profiler_command_payload_get_attach_timeout`).

Reply:
Ah. I could not figure out where `ds_attach_profiler_command_payload_get_attach_timeout` comes from. It didn't show up anywhere else?

Looks like I should therefore not remove any of the timeout code, right?

Reply:
done